packages/core/src/plugins/meta/discogs.ts

```diff
@@ -146,10 +150,22 @@ class DiscogsMetaProvider extends MetaProvider {
   }> {
     return Discogs.search(query)
       .then(response => response.json())
-      .then(json => json.map(result => ({
-        ...result,
-        source: SearchResultsSource.Discogs
-      })));
+      .then(json => {
+        const artists = [];
+        const releases = [];
+
+        if (json.results) {
+          json.results.map(item => {
```

Comment:
instead of pushing inside map, just assign the maps to consts (it's fine to map twice)

Reply:
How to do it? I haven't done it before.

Reply:
how ab this? cos  map return "undefined" if we don't return any
```js
  if (json.results) {
    const artists = json.results.flatMap(item => 
      (item.type === 'artist') ?
        [this.discogsArtistSearchResultToGeneric(item)] : []
    );

    const releases = json.results.flatMap(item =>
      (item.type === 'master' || item.type === 'release' ) ?
        [this.discogsReleaseSearchResultToGeneric(item)] : []
    );

    return Promise.resolve({ artists, releases, tracks: [] });
  }

  return Promise.resolve({ artists: [], releases: [], tracks: [] });
```

Reply:
Yes, this is great.

In some situations you can use this: https://lodash.com/docs/4.17.15#compact

But your approach is better here